ui/src/components/App/App.jsx

```diff
@@ -1,13 +1,77 @@
 import { useState, useEffect } from 'react'
+import * as React from "react"
 import './App.css'
 import NavBar from '../NavBar/NavBar'
 import LoggedOutView from '../LoggedOutView/LoggedOutView'
 import MessagesView from '../MessagesView/MessagesView'
 import axios from "axios"
 import {BrowserRouter,Routes,Route} from "react-router-dom"
+import * as config from '../../config'
 
-export default function App() {
-  const [isLoggedIn, setIsLoggedIn] = useState(localStorage.getItem("current_user_id") !== null)
+export default function App() { 
+  const [isLoggedIn, setIsLoggedIn]         = useState(localStorage.getItem("current_user_id") !== null)
+  const [city, setCity]                     = useState("");
+  const [state, setState]                   = useState("");
+  const [listings, setListings]              = useState([]);
+  const [properties, setProperties]         = useState([]);
+  const [average, setAverage]               = useState(0);
+  const [minPrice, setMinPrice]             = useState(0);
+  const [maxPrice, setMaxPrice]             = useState(0);
+  const [numProperties, setNumProperties]   = useState(0);
+  
+  React.useEffect(() => {
+    // Temp variables used to calculate the average rent price in a city and the
+    // number of properties in the user's price range
+     let totalPrice      = 0;
+     let totalProperties = 0;
+
+     // Iterate over all listings in the city, add the price of each listing to 
+     // calculate the average
+     listings.forEach((listing) => {
+      totalPrice += listing.price;
+
+      // If the given listing is within the user's price range, add the properties to
+      // the array of properties within the user's price range
+      // If the given listing is within the user's price range, increment the counter
+      // variable for the number of properties within the user's price range 
+      if (listing.price >= minPrice && listing.price <= maxPrice) {
+        ++totalProperties;
+        setProperties((prev) => [...prev, listing]);
+        console.log(properties);
+        console.log(listing);
+      }
+     });
+
+     totalPrice /= listings.length;
+
+     // Update state variables
+     setAverage(totalPrice);
+     setNumProperties(totalProperties);
```

Comment:
you will probably be doing more initialization in the App.jsx file later. In order for it to not get as crowded, it would be worth it to separate this logic into a separate function - that way, you can also re-use it later to recalculate average price let's say if the user changes it in the profile view. I'd suggest creating a file called "Utils.js" where we would store any computation/business logic information and then you can name a function called "calculate average price" and "calculate total properties" as separate functions

Reply:
Good idea!!